hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,246 @@
+            periodSeconds = defaultValue;
+        }
+
+        registerStatistics();
+
+        schedulePeriodicStatisticsSendTask(periodSeconds);
+
+        logger.info("Client statistics is enabled with period " + periodSeconds + " seconds.");
+    }
+
+    private ClientConnection getOwnerConnection() {
+        Address ownerConnectionAddress = client.getClientClusterService().getOwnerConnectionAddress();
+        if (null == ownerConnectionAddress) {
+            return null;
+        }
+        ClientConnection connection = (ClientConnection) client.getConnectionManager().getConnection(ownerConnectionAddress);
+        if (null == connection) {
+            return null;
+        }
+
+        int serverVersion = connection.getConnectedServerVersion();
+        if (serverVersion < FEATURE_SUPPORTED_SINCE_VERSION) {
+            if (logger.isFinestEnabled()) {
+                logger.finest(format("Client statistics can not be started since current connected owner server version is less than"
+                        + " the minimum supported server version %s", FEATURE_SUPPORTED_SINCE_VERSION_STRING));
+            }
+            return null;
+        }
+
+        return connection;
+    }
+
+    private void schedulePeriodicStatisticsSendTask(long periodSeconds) {
+        client.getExecutionService().scheduleWithRepetition(new Runnable() {
+            @Override
+            public void run() {
+                ClientConnection ownerConnection = getOwnerConnection();
+                if (null == ownerConnection) {
+                    logger.finest("Can not send client statistics to the server. No owner connection.");
+                    return;
+                }
+
+                final Map<String, String> stats = new HashMap<String, String>();
+
+                stats.put("lastStatisticsCollectionTime", Long.toString(System.currentTimeMillis()));
+
+                stats.putAll(staticStats);
+
+                periodicStats.fillMetrics(stats);
+
+                ThreadPoolExecutor userExecutor = (ThreadPoolExecutor) client.getClientExecutionService().getUserExecutor();
+                stats.put("userExecutor.queueSize", Long.toString(userExecutor.getQueue().size()));
+
+                addNearCachStats(stats);
+
+                sendStats(stats, ownerConnection);
+            }
+        }, 0, periodSeconds, SECONDS);
+    }
+
+    private void addNearCachStats(Map<String, String> stats) {
+        for (NearCache nearCache : client.getNearCacheManager().listAllNearCaches()) {
+            String pathPrefix = NEAR_CACHE_CATEGORY_PREFIX;
+            String name = nearCache.getName();
+            if (name.startsWith("/")) {
+                pathPrefix += name.substring(1);
+            } else {
+                pathPrefix += name;
+            }
+            pathPrefix += ".";
+
+            NearCacheStats nearCacheStats = nearCache.getNearCacheStats();
+
+            StringBuilder buffer = new StringBuilder();
```

Comment:
Why is the buffer created in the loop? If you create it outside the loop, you get less litter.